calabash-cucumber/test/cucumber/features/steps/device_agent_api.rb

```diff
@@ -83,3 +83,90 @@ def expect_device_agent_to_screenshot(method, uiquery=nil)
 When(/^device_agent\.enter_text_in fails, it generates a screenshot$/) do
   expect_device_agent_to_screenshot(:enter_text_in)
 end
+
+Then(/^I can open Notifications tab in Today and Notifications page$/) do
+  if ipad?
+    puts "Test is not stable on iPad; skipping"
+  end
+
+  element = wait_for_view("*")
+  x = element["rect"]["center_x"]
+  final_y = element["rect"]["center_y"] + (element["rect"]["height"]/2)
+  pan_coordinates({:x => x, :y => 0},
+                  {:x => x, :y => final_y},
+                  {duration: 0.5})
+
+  # Waiting for animations is not good enough - the animation is outside of
+  # the AUT's view hierarchy
+  wait_for_external_animations
+
+  # Screenshots will not show the Control Panel page.
+  if ios9?
+    if uia_available?
+      if uia_call_windows([:button, {marked: 'Notifications'}], :isVisible) != 1
+        fail("Expected to see 'Notifications' element.")
+      end
+
+      uia_tap :button, marked: 'Notifications'
+      wait_for_external_animations
+
+      if uia_call_windows([:element, {marked: 'No Notifications'}], :isVisible) != 1
+        fail("Expected to see 'No Notifications' element")
+      end
+    else
+      wait_for(options_or_timeout=3) do
+        !device_agent.query({marked: 'Notifications'}).empty?
+      end
+
+      device_agent.touch({:marked => 'Notifications'})
+
+      wait_for(options_or_timeout=3) do
+        !device_agent.query({marked: 'No Notifications'}).empty?
+      end
+    end
+  elsif ios10?
+    wait_for(options_or_timeout=3) do
+      !device_agent.query({marked: 'No Notifications'}).empty?
+    end
+  end
+end
+
+Then(/^I can see Control Panel page elements$/) do
+  if ipad?
+    puts "Test is not stable on iPad; skipping"
+  end
```

Comment:
This will not skip the iPad; it will continue the test.

You cannot return from the `do` block, so you will have to wrap the whole step in `if !ipad?`.

Did I make that mistake too in my step?